modules/openapi-generator/src/main/resources/python/api_client.mustache

```diff
@@ -186,22 +186,43 @@ class ApiClient(object):
             # use server/host defined in path or operation instead
             url = _host + resource_path
 
-        # perform request and return response
-        response_data = {{#asyncio}}await {{/asyncio}}{{#tornado}}yield {{/tornado}}self.request(
-            method, url, query_params=query_params, headers=header_params,
-            post_params=post_params, body=body,
-            _preload_content=_preload_content,
-            _request_timeout=_request_timeout)
-
-        self.last_response = response_data
+        try:
+            # perform request and return response
+            response_data = {{#asyncio}}await {{/asyncio}}{{#tornado}}yield {{/tornado}}self.request(
+                method, url, query_params=query_params, headers=header_params,
+                post_params=post_params, body=body,
+                _preload_content=_preload_content,
+                _request_timeout=_request_timeout)
+
+            content_type = response_data.getheader('content-type')
+
+            self.last_response = response_data
+
+            return_data = response_data
+
+            if not _preload_content:
+                {{^tornado}}
+                return return_data
+                {{/tornado}}
+                {{#tornado}}
+                raise tornado.gen.Return(return_data)
+                {{/tornado}}
+
+            if six.PY3 and response_type not in ["file", "bytes"]:
+                match = None
+                if content_type is not None:
+                    match = re.search(r"charset=([a-zA-Z\-\d]+)[\s\;]?", content_type)
+                encoding = match.group(1) if match else "utf-8"
+                response_data.data = response_data.data.decode(encoding)
 
-        return_data = response_data
-        if _preload_content:
             # deserialize response data
             if response_type:
                 return_data = self.deserialize(response_data, response_type)
             else:
                 return_data = None
+        except ApiException as e:
+            e.body = e.body.decode('utf-8') if six.PY3 else e.body
+            raise e
```

Comment:
Is this only raised by the `self.request` call?
If so can we move this up to be right after that call?
That way all of the other code can be moved out of this try block because it doesn't need to be inside it.